addons/main.go

```diff
@@ -188,3 +198,62 @@ func setupChecks(mgr ctrl.Manager) {
 		os.Exit(1)
 	}
 }
+
+func enableClusterBootstrapAndConfigControllers(ctx context.Context, mgr ctrl.Manager) {
+	if err := (&calicocontroller.CalicoConfigReconciler{
+		Client: mgr.GetClient(),
+		Log:    ctrl.Log.WithName("CalicoConfigController"),
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(ctx, mgr, controller.Options{MaxConcurrentReconciles: 1}); err != nil {
+		setupLog.Error(err, "unable to create CalicoConfigController", "controller", "calico")
+		os.Exit(1)
+	}
+
+	if err := (&antreacontroller.AntreaConfigReconciler{
+		Client: mgr.GetClient(),
+		Log:    ctrl.Log.WithName("AntreaConfigController"),
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(ctx, mgr, controller.Options{MaxConcurrentReconciles: 1}); err != nil {
+		setupLog.Error(err, "unable to create AntreaConfigController", "controller", "antrea")
+		os.Exit(1)
+	}
+	if err := (&kappcontroller.KappControllerConfigReconciler{
+		Client: mgr.GetClient(),
+		Log:    ctrl.Log.WithName("KappControllerConfig"),
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(ctx, mgr, controller.Options{MaxConcurrentReconciles: 1}); err != nil {
+		setupLog.Error(err, "unable to create KappControllerConfig", "controller", "kapp")
+		os.Exit(1)
+	}
+	if err := (&cpicontroller.CPIConfigReconciler{
+		Client: mgr.GetClient(),
+		Log:    ctrl.Log.WithName("CPIConfig"),
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(ctx, mgr, controller.Options{MaxConcurrentReconciles: 1}); err != nil {
+		setupLog.Error(err, "unable to create KappControllerConfig", "controller", "kapp")
+		os.Exit(1)
+	}
+
+	bootstrapReconciler := controllers.NewClusterBootstrapReconciler(
+		mgr.GetClient(),
+		ctrl.Log.WithName("ClusterBootstrapController"),
+		mgr.GetScheme(),
+		&addonconfig.ClusterBootstrapControllerConfig{
+			CNISelectionClusterVariableName: constants.DefaultCNISelectionClusterVariableName,
+			HTTPProxyClusterClassVarName:    constants.DefaultHTTPProxyClusterClassVarName,
+			HTTPSProxyClusterClassVarName:   constants.DefaultHTTPSProxyClusterClassVarName,
+			NoProxyClusterClassVarName:      constants.DefaultNoProxyClusterClassVarName,
+			ProxyCACertClusterClassVarName:  constants.DefaultProxyCaCertClusterClassVarName,
+			IPFamilyClusterClassVarName:     constants.DefaultIPFamilyClusterClassVarName,
+			SystemNamespace:                 constants.TKGSystemNS,
+			PkgiServiceAccount:              constants.PackageInstallServiceAccount,
```

Comment:
question: do we want to use the value from a flag var (`addonServiceAccount`) ? The same might apply to the following few lines.

Reply:
We made service account, etc configurable for exisint addons-manager and never really saw use case. I think the idea is needs to be unique within the namespace and I prefixed with `tanzu-cluster-bootstrap-` which should be pretty unique. I suggest we revisit making configurable if we really run into it during testing.